nc2zarr/writer.py

```diff
@@ -172,3 +193,89 @@ def _remove_variable_attrs(cls, ds: xr.Dataset) -> xr.Dataset:
         for k, v in ds.variables.items():
             v.attrs = dict()
         return ds
+
+    def _finalize_dataset(self):
+        with log_duration('Finalizing dataset'):
+            adjusted_metadata = {}
+
+            if self._output_adjust_metadata:
+                self._ensure_store()
+                # Get new attribute values
+                with xr.open_zarr(self._output_store, decode_cf=True) as dataset:
+                    history = self._get_history_metadata(dataset)
+                    source = self._get_source_metadata(dataset)
+                    time_coverage_start, time_coverage_end = \
+                        self._get_time_coverage_metadata(dataset)
+                    adjusted_data = (
+                        ('history', history),
+                        ('source', source),
+                        ('time_coverage_start', time_coverage_start),
+                        ('time_coverage_end', time_coverage_end),
+                    )
+                    adjusted_metadata = {k: v
+                                         for k, v in adjusted_data
+                                         if v is not None}
+            if self._output_metadata:
+                adjusted_metadata.update(self._output_metadata)
+
+            LOGGER.info(f'Metadata update:\n'
+                        f'{json.dumps(adjusted_metadata, indent=2)}')
+
+            if not self._dry_run:
+                if adjusted_metadata:
+                    self._ensure_store()
+                    # Externally modify attributes
+                    with zarr.open_group(self._output_store,
+                                         cache_attrs=False) as group:
+                        group.attrs.update(adjusted_metadata)
+                if self._output_consolidated:
```

Comment:
I see a potential danger here: if the `consolidated` option was not specified on this run, but the Zarr store itself *is* consolidated, this will (I believe) leave the store in an inconsistent state, with the metadata updates not propagated to the consolidated object. So we need an `... or group.is_consolidated()` added to the `if` clause, expect that this method doesn't exist, but `".zmetadata" in group.store` should work for this purpose.

Reply:
I don't agree, at least not for this PR. It is up to configuration to specify whether consolidation should take place or not.

Reply:
Please open an issue later, should this really become a problem.

Reply:
I only partly agree. If you want to consolidate, you should configure nc2zarr to do so, otherwise nc2zarr could later delete an existing `.zmetadata` or emit a warning. Open an feature request later once this becomes a real problem for us.

Reply:
I changed my mind: the new default behavior is: if `consolidated` is not specified or is False AND there are metadata updates, then consolidate. 

Actually this is only valid if consolidated is not specified at all, namely `consolidated is None`. If it is specified to be `consolidated is False` then an existing `.zmetadata` shall be deleted. Will make this an issue later.